shiny/ui/_theme_brand.py

```diff
@@ -0,0 +1,297 @@
+        "weight": "headings-font-weight",
+        "color": "headings-color",
+        "style": "headings-style",
+    },
+    "monospace": {
+        "family": "font-family-monospace",
+        "size": "code-font-size",
+    },
+    "monospace_inline": {
+        "family": "font-family-monospace-inline",
+        "color": "code-color",
+        "background_color": "code-bg",
+        "size": "code-inline-font-size",
+        "weight": "code-font-weight",
+    },
+    "monospace_block": {
+        "family": "font-family-monospace-block",
+        "line_height": "pre-line-height",
+        "color": "pre-color",
+        "background_color": "pre-bg",
+    },
+    "link": {
+        "background_color": "link-bg",
+        "color": "link-color",
+        "weight": "link-weight",
+        "decoration": "link-decoration",
+    },
+}
+"""Maps brand.typography fields to corresponding Bootstrap Sass variables"""
+
+
+class BrandBootstrap:
+    """Convenience class for storing Bootstrap defaults from a brand instance"""
+
+    def __init__(
+        self,
+        version: Any = bootstrap,
+        preset: Any = "shiny",
+        **kwargs: str | int | bool | float | None,
+    ):
+        if not isinstance(version, (str, int)):
+            raise ValueError(
+                f"Bootstrap version must be a string or integer, not {version!r}."
+            )
+
+        v_major = str(version).split(".")[0]
+        bs_major = str(bootstrap).split(".")[0]
+
+        if v_major != bs_major:
+            # TODO (bootstrap-update): Assumes Shiny ships one version of Bootstrap
+            warnings.warn(
+                f"Shiny does not current support Bootstrap version {v_major}. "
+                f"Using Bootstrap v{bs_major} instead.",
+                stacklevel=4,
+            )
+            v_major = bs_major
+
+        if not isinstance(preset, str) or preset not in shiny_theme_presets:
+            raise ValueError(
+                f"{preset!r} is not a valid Bootstrap preset provided by Shiny. "
+                f"Valid presets are {shiny_theme_presets}."
+            )
+
+        self.version = v_major
+        self.preset: ShinyThemePreset = preset
+        self.defaults = kwargs
+
+    @classmethod
+    def from_brand(cls, brand: Brand):
+        defaults: dict[str, str | int | bool | float | None] = {}
+
+        if brand.defaults:
+            if brand.defaults and "bootstrap" in brand.defaults:
+                defaults.update(brand.defaults["bootstrap"])
+            if "shiny" in brand.defaults and "theme" in brand.defaults["shiny"]:
+                defaults.update(brand.defaults["shiny"]["theme"])
+
+        return cls(**defaults)
+
+
+class ThemeBrand(Theme):
+    def __init__(
+        self,
+        brand: Brand,
+        *,
+        include_paths: Optional[str | Path | list[str | Path]] = None,
+    ):
+        if not isinstance(brand, Brand):
+            raise ValueError("Invalid `brand`, must be a path or a Brand instance.")
```

Comment:
I'd also be ok with removing these runtime type checks since we are fully in control of the input (and thus, we can just rely on the type checking)

```suggestion
            raise ValueError("Invalid `brand`, must a Brand instance.")
```